### PR TITLE
Add property to enable/disable pruning at runtime

### PR DESCRIPTION
**Pull request by rhaschke**
Rebase and cleanup of #358. Closes #358.

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -40,9 +40,7 @@ catkin_package(
 )
 
 add_compile_options(-fvisibility-inlines-hidden)
-if(ENABLE_PRUNING)
-	add_compile_definitions(ENABLE_PRUNING)
-endif()
+add_compile_definitions(ENABLE_PRUNING=$<BOOL:${ENABLE_PRUNING}>)
```
Comment (sjahr) [inline core/CMakeLists.txt:43]: What is that doing?

File: core/src/task.cpp [hunk 2]
```diff
@@ -92,6 +92,7 @@ const ContainerBase* TaskPrivate::stages() const {
 
 Task::Task(const std::string& ns, bool introspection, ContainerBase::pointer&& container)
   : WrapperBase(new TaskPrivate(this, ns), std::move(container)) {
+	setPruning(ENABLE_PRUNING);
```
Comment (sjahr) [inline core/src/task.cpp:95]: Wouldn't it be possible to just have a C++ API with defaulting to true without exposing this via CMake?

**Review by sjahr — COMMENTED** [review]
I like the idea of having a simple property based API around this.I am not fully understanding the macros around this, is it necessary to have a macro-based API around this?

File: core/CMakeLists.txt (hunk 1 above)
Comment (rhaschke) [reply core/CMakeLists.txt:43]: This is exposing the cmake option as a compiler definition to serve as the default of `setPruning()` below.

**Pull request merged**